src/folders.ts

```diff
@@ -79,10 +80,19 @@ export class CMakeToolsFolderController implements vscode.Disposable {
 
   /**
    * Get the CMakeTools instance associated with the given workspace folder, or undefined
-   * @param ws The workspace folder to search
+   * @param ws The workspace folder to search, or array of command and workspace path
    */
-  get(ws: vscode.WorkspaceFolder | undefined): CMakeToolsFolder | undefined {
-    return ws ? this._instances.get(ws.uri.fsPath) : ws;
+  get(ws: vscode.WorkspaceFolder | Array<string> | undefined): CMakeToolsFolder | undefined {
+    if (ws) {
+      if (util.isArrayOfString(ws)) {
+        const command = ws as Array<string>;
+        return this._instances.get(command[command.length - 1]);
+      } else if ((ws as vscode.WorkspaceFolder).uri) {
```

Comment:
You should use `ws instanceof vscode.WorkspaceFolder` to check the type or just test the existence of `ws.uri` directly. The `as` type assertion doesn't do what you want.

Reply:
Since `vscode.WorkspaceFolder` is interface, instanceof is not working with it.

Reply:
Sorry, just've noticed... `assertion`. I've seen practices doing exactly what I wanted this way: check type and field at the same time. Field check isn't actually as necessary here as type check - but it seems I need something differ from `instanceof` anyway.

Reply:
You're right.  That's what we have `util.isArrayOfString` for. You'd need to write another type assertion function like that one that checks that the required properties exist and are of the correct type.  In this case, however, you are checking that the `uri` property exists explicitly, so I suppose that no type assertion is actually needed here.

Reply:
Ended up with `util.isWorkspaceFolder`.